### PR TITLE
docs: Refactor "Generating components" page

### PR DESCRIPTION
**Pull request by najikahalsema**
## Description

Continuing my editing of our onboarding docs... Changed typos and tried to make things clearer. 

## Motivation and Context

Better docs :^) 

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: documentation/guides/generating-components.md [hunk 1]
```diff
@@ -28,10 +28,12 @@ In response to the above, the component name should be the dashified version of
    package.json
 ```
 
-From here, peek into the `package.json` file and ensure the "devDependency" of `@spectrum-css/spectrumpattern` listed there is of the most current version. Once you have confirmed that run `yarn` in your terminal to get any new packages added to the repository and have the CSS provided therein processed for use in your component using the default configuration. You'll now be able to review your component via the Storybook command `yarn storybook` or see it appear in tests via `yarn test` so that you can dive into fully preparing the patterns for use as a custom element.
+From here, peek into the `package.json` file and ensure the "devDependency" of `@spectrum-css/spectrumpattern` listed there is of the most current version. Then, run `yarn` in your terminal to grab any newly-added packages, as well as to ensure that you have the provided CSS processed for use in your component. You'll now be able to see your component in the Storybook, using the command `yarn storybook`, or test its functionality via `yarn test` so that you can dive into fully preparing the pattern for use as a custom element.
```
Comment (Westbrook) [inline documentation/guides/generating-components.md:31]: Everything through
```
From here, peek into the `package.json` file and ensure the "devDependency" of `@spectrum-css/spectrumpattern` listed there is of the most current version. Then, run `yarn` in your terminal to grab any newly-added packages, as well as to ensure that you have the provided CSS processed for use in your component.
```
has changed a little bit.

Do you want to checkout the workflow as it is now and edit this PR, or move this one along and create a follow up that updates the workflow of this `yarn new-package` command?

File: documentation/guides/generating-components.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+The fastest way to get started with creating a new component is to generate it from the command line. Run the following command to have your new component scaffolded for you:
 
 ```bash
-yarn new-package
+$ yarn new-package
 ? component name please _
 ```
 
-In response to the above, the component name should be the dashified version of the Spectrum CSS pattern that you are looking to add to the repo. In this way, the `SpectrumPattern` that you want to create will be represented as `spectrum-pattern` in your response. From here the command will generate a new package for you pattern with the following file structure in the `packages` directory:
+In response to the prompt above, the component name should be the kebab case version of the Spectrum CSS pattern that you are looking to add to the repo. That is, the `SpectrumPattern` that you want to create should be represented as `spectrum-pattern` in your response. From here, the command generates a new package for your pattern with the following file structure in the `/packages` directory:
```
Comment (Westbrook) [inline documentation/guides/generating-components.md:10]:
```suggestion
In response to the prompt above, the component name should be the kebab case version of the Spectrum CSS pattern that you are looking to add to the repo. The `SpectrumPattern` that you want to create should be represented as `spectrum-pattern` in your response. From here, the command generates a new package for your pattern with the following file structure in the `/packages` directory:
```
Maybe, just...

File: documentation/guides/generating-components.md [hunk 3]
```diff
@@ -28,10 +28,12 @@ In response to the above, the component name should be the dashified version of
...
+The next place to look is in `node_modules/@spectrum-css/spectrumpattern/metadata/spectrumpattern.yml`. Here, you will find complete Spectrum CSS’s HTML representation of the many states, variants, and capabilities offered by the pattern that you are working with. The content of this file is also found on the <sp-link href="https://opensource.adobe.com/spectrum-css">Spectrum CSS documentation site</sp-link> under the name of the pattern you’re leveraging. Following this as a guide while you implement this component ensures that you will fulfill the expected features of this pattern.
```
Comment (Westbrook) [inline documentation/guides/generating-components.md:33]:
```suggestion
The next place to look is in `node_modules/@spectrum-css/spectrumpattern/metadata/spectrumpattern.yml`. Here, you will find Spectrum CSS’s HTML representation of the many states, variants, and capabilities offered by the pattern that you are working with. The content of this file is also found on the <sp-link href="https://opensource.adobe.com/spectrum-css">Spectrum CSS documentation site</sp-link> under the name of the pattern you’re developing. Following this as a guide while you implement this component ensures that you will fulfill the expected features of this pattern.
```

File: documentation/guides/generating-components.md [hunk 4]
```diff
@@ -28,10 +28,12 @@ In response to the above, the component name should be the dashified version of
...
+Now that you have this scaffold as a base, check out <sp-link href="guides/developing-components">these instructions</sp-link> on working with web components in general, and the <sp-link href="guides/spectrum-config">annotated `spectrum-config.js`</sp-link> file, which outlines how to translate the Spectrum CSS source into shadow DOM-optimized CSS. You are well on your way to adding a new component package to the project.
 
-Thanks for stopping by, we look forward to your contribution!
+If you run into any issues with the instructions above, or the ones linked across this documentation site, please feel free to raise questions or issues on <sp-link href="https://github.com/adobe/spectrum-web-components/issues">GitHub</sp-link>. You can do this either by creating a brand new issue when it specifically relates to the process of generating a component, or by leaving a comment on an already-created issue for a specific pattern when it concerns bringing that component to life.
```
Comment (Westbrook) [inline documentation/guides/generating-components.md:37]: This is great. Should be enable the "Dicussions" feature for this? https://docs.github.com/en/discussions

**Review by Westbrook — CHANGES_REQUESTED** [review]

File: documentation/guides/generating-components.md (hunk 1 above)
Comment (najikahalsema) [reply documentation/guides/generating-components.md:31]: Hmm. I might just push this one for now and create a follow-up PR tomorrow. Thank you for letting me know, I figured there'd be some deprecated stuff in here :)

File: documentation/guides/generating-components.md (hunk 4 above)
Comment (najikahalsema) [reply documentation/guides/generating-components.md:37]: I think that would be a great thing for SWC!

Comment (Westbrook) [reply documentation/guides/generating-components.md:37]: Feel free to do this in the follow up work, with the most recent `yarn new-package` workflow, but we should also include this here: https://github.com/adobe/spectrum-web-components/discussions for `raise questions`.

**Review by Westbrook — CHANGES_REQUESTED** [review]

**Review by Westbrook — DISMISSED** [review]

**Review by Westbrook — APPROVED** [review]

**Pull request merged**